report/sections/Frontend/chapter.tex

```diff
@@ -0,0 +1,86 @@
+\chapter{Frontend} \label{chap:Frontend}
+One of the primary goals of this application to give the lecturer a way to specify what constitutes a correct problem solution.
+As previously mentioned in the \textbf{Use Cases} section of chapter \ref{chap:Specification}, a common approach to this is the use of test cases.
+
+\section{Authentication and Roles}
+When a user initially loads the frontend website, they will not be signed in, and will will only have minimal privileges.
+This means that they will not have access to any of the course material or exercises.
+In order to access any of these features, the user has to sign in.
+By pressing the sign in button, the user is redirected to a sign in page that is specific to the authentication provider used.
+In our case, it is Google.
+After logging in with the authentication provider, the user is redirected back to the website.
+At this point, the user is now logged in and our database will be updated with the user's information, as well as a token to keep track of the user's session.
+This token is used by NextAuth to create a session for the user, which will be used to authenticate the user for subsequent requests to the site.
+The session also has a role associated with it, which determines what actions the user is allowed to take on the site.
+
+\begin{figure}[H]
+    \centering
+    \frame{\includegraphics[scale=0.1]{home.png}}
+    \caption{Home page as a logged-in user}
+    \label{fig:Home page}
+\end{figure}
+
+\section{Syllabi, Problem sets and Exercises}
+On the website, each page that displays course data makes a request to a Next.js backend API endpoint to retrieve the data from the database.
+This request is made using tRPC, which allows for efficient communication between the frontend and the backend. tRPC wraps around React Query, which provides useful features such as caching and automatic refetching.
+
+The request input is validated using zod to ensure that it is in the correct format.
```

Comment:
After the full stop, maybe explain briefly what zod does and why we use it.

Reply:
Removed Zod by request of @TheRumle